src/AuthoringRunnerServiceHandler.cs

```diff
@@ -111,12 +111,12 @@ public override Task<Empty> Kill(KillProcessRequest request, ServerCallContext c
             try
             {
                 Logger.Debug("KillProcessrequest received");
-                lifetime.StopApplication();
                 return Task.FromResult(new Empty());
             }
             finally
             {
                 _pool.Dispose();
+                Task.Delay(500).ContinueWith((_) => lifetime.StopApplication());
```

Comment:
From https://github.com/grpc/grpc-dotnet/issues/1649#issuecomment-1081854085

> This is a pretty low-level change in HTTP/2. Given the risk, for now, the fix will be in .NET 7 and later. We'll reevaluate backporting if more people are impacted by this.

Maybe it's worth waiting for the fix?